### ISSUE TITLE
Datafusion parses interval literal error

### ISSUE THREAD

**Issue by goldmedal** [Canner/wren-engine#768]
**Describe the bug**

Found from TPCH test.
```
DataFusion error: Arrow error: Parser error: Invalid input syntax for type interval: "0 YEARS 3 MONS 0 DAYS 0 HOURS 0 MINS 0.000000000 SECS seconds"
```

DataFusion will unparse an interval value to a full interval literal but Datafusion can't parse this result.

**To Reproduce**

https://github.com/Canner/wren-engine/blob/32f3f63cf083e309d780553e3c7835e7b5372922/wren-modeling-rs/sqllogictest/test_files/tpch/q15.slt.part#L19-L21

**Comment by goldmedal** [comment]
It would be fixed by https://github.com/apache/arrow-rs/pull/6006 and https://github.com/apache/datafusion/pull/12032

**Comment by goldmedal** [comment]
solved by #802
